server/src/main/java/org/candlepin/model/activationkeys/ActivationKeyCurator.java

```diff
@@ -33,9 +33,15 @@ public ActivationKeyCurator() {
         super(ActivationKey.class);
     }
 
-    public List<ActivationKey> listByOwner(Owner owner) {
-        return (List<ActivationKey>) currentSession().createCriteria(ActivationKey.class)
-        .add(Restrictions.eq("owner", owner)).list();
+    public CandlepinQuery<ActivationKey> listByOwner(String keyName, Owner owner) {
```

Comment:
Bit of a stylistic thing here, but try to keep the parameter order in order of significance. Owner is operative here (as we're listing by owner) and required, so it should be first in line. keyName being optional definitely goes second. For bonus points, you could overload the method with another definition that leaves off the keyName and passes null through for it.

```
public ... listByOwner(Owner owner) {
    return this.listByOwner(owner, null);
}
```